std/math.d

```diff
@@ -6302,34 +6303,34 @@ version (D_HardFloat) @safe unittest // rounding
  * Returns:
  *  `true` if $(D_PARAM x) is Nan.
  */
-bool isNaN(X)(X x) @nogc @trusted pure nothrow
+bool isNaN(X)(const X x) @nogc @trusted pure nothrow
```

Comment:
`const` here causes an error in CTFE for `isNaN(real.nan)`. Removing the `const` allows it to work fine. Bizarre.

Reply:
My guess is that `isNaN` (along with other functions) is evaluated as a built-in at compile-time and such built-ins are matched by their mangled name.

Reply:
Yep. Although some of those builtins are outdated and can (will!) removed at some point, because the real builtins are in `core`.